smarty/sampler.py

```diff
@@ -129,41 +144,16 @@ def __init__(self, molecules, basetypes_filename, initialtypes_filename, decorat
                 self.atomtypes = [(smarts, typename)] + self.atomtypes
```

Comment:
I've been trying to think through if adding in basetypes in this way makes sense. I'm a little concerned that we're getting things out of order. 

So I think what should happen is that you add any "missing" basetypes to the top of the initial atomtype list in the same order as they appeared in your basetype list. For example if you had basetypes:
```
[*] any
[$ewg1] ewg1
[#6] C
[#8] O
[#1] H
```
and initial types:
```
[#1] hydrogen
[#6] carbon
[#8] oxygen
```
The final list should be:
```
[*] any
[$ewg1] ewg1
[#1] hydrogen
[#6] carbon
[#8] oxygen
```
In the current form I think you would get 
```
[$ewg1] ewg1
[*] any
[#1] hydrogen
[#6] carbon
[#8] oxygen
```

@camizanette I'm still thinking about the best way to avoid this problem, thoughts?